[RunjiaChen] iP

RunjiaChen opened the pull request:
# Snom Bot :grinning:

> We must all suffer from one of two pains: the pain of discipline or the pain of regret. 

List of features:
- recording tasks
- marking tasks as done 
- list down tasks


Steps to run * *Snombot * *:
1. download the jar file from the website [my repo](https://github.com/RunjiaChen/ip)
2. run the jar file in terminal 

So, are you ready to use Snombot? 
- [ ] Yes
- [ ]  No

If you're learning java, you can experiment with the code too. Here is the `main` method 

```java 
public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

RunjiaChen committed:
test

RunjiaChen committed:
no message

RunjiaChen committed:
Level-0

RunjiaChen committed:
Level-1

RunjiaChen committed:
no message

RunjiaChen committed:
no message

RunjiaChen committed:
Level 3

RunjiaChen committed:
Level 4

RunjiaChen committed:
no message

RunjiaChen committed:
Level 5

RunjiaChen committed:
no message

RunjiaChen committed:
Level-6

RunjiaChen committed:
Merge branch 'TestBranch'

# Conflicts:
#	src/main/java/Duke.java

RunjiaChen committed:
no message

RunjiaChen committed:
no message

RunjiaChen committed:
no message

RunjiaChen committed:
Added Level-7 features

RunjiaChen committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Added Level-7 features

RunjiaChen committed:
Add date time features

RunjiaChen committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Add date time features

RunjiaChen committed:
Added Command Classes

RunjiaChen committed:
Added Exceptions

RunjiaChen committed:
Add TaskList

RunjiaChen committed:
Merge commit '2be85b94650fa3fea90f646248147f68af6ab985'

* commit '2be85b94650fa3fea90f646248147f68af6ab985':
  Added Command Classes

RunjiaChen committed:
Merge commit '13293fb1fad25f7c7c7a85c0bd71f42693200664'

* commit '13293fb1fad25f7c7c7a85c0bd71f42693200664':
  Added Exceptions

RunjiaChen committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Add TaskList

RunjiaChen committed:
Add more Exceptions

RunjiaChen committed:
Add more commands

RunjiaChen committed:
Add TaskList Class

RunjiaChen committed:
Add AddEventCommand, AddTodoCommand, AddDeadlineCommand

RunjiaChen committed:
Add more task types

RunjiaChen committed:
Add TaskStorage

RunjiaChen committed:
Add TaskList

RunjiaChen committed:
Add Ui

RunjiaChen committed:
Add enums

RunjiaChen committed:
Add more classes

RunjiaChen committed:
Merge branch 'branch-A-MoreOOP'

* branch-A-MoreOOP:
  Add more classes
  Add enums
  Add Ui
  Add TaskList
  Add TaskStorage
  Add more task types
  Add AddEventCommand, AddTodoCommand, AddDeadlineCommand
  Add TaskList Class
  Add more commands
  Add more Exceptions

RunjiaChen committed:
Merge branch 'master' into add-gradle-support

* master: (30 commits)
  Add more classes
  Add enums
  Add Ui
  Add TaskList
  Add TaskStorage
  Add more task types
  Add AddEventCommand, AddTodoCommand, AddDeadlineCommand
  Add TaskList Class
  Add more commands
  Add more Exceptions
  Add TaskList
  Added Exceptions
  Added Command Classes
  Add date time features
  Added Level-7 features
  no message
  no message
  no message
  Level-6
  no message
  ...

RunjiaChen committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  Bump gradle and lib version
  Add Gradle support
  docs/README.md: Tweak document template

RunjiaChen committed:
Add JUnit for Todo and Deadline

RunjiaChen committed:
Merge branch 'branch-A-JUnit'

* branch-A-JUnit:
  Add JUnit for Todo and Deadline

RunjiaChen committed:
Add enums

RunjiaChen committed:
Merge branch 'branch-A-JUnit'

* branch-A-JUnit:
  Add enums

RunjiaChen committed:
Add Jar file parameters

RunjiaChen committed:
Merge branch 'branch-A-Jar'

* branch-A-Jar:
  Add Jar file parameters

RunjiaChen committed:
Add Javadoc for classes and methods

RunjiaChen committed:
Add find command

RunjiaChen committed:
Add checkstyle and coding standard

RunjiaChen committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Add checkstyle and coding standard

RunjiaChen committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add Javadoc for classes and methods

# Conflicts:
#	src/main/java/inputcommands/AddDeadlineCommand.java
#	src/main/java/inputcommands/AddTodoCommand.java
#	src/main/java/inputcommands/MarkTaskCommand.java
#	src/main/java/inputcommands/UnmarkTaskCommand.java
#	src/main/java/snomexceptions/InvalidCommandDateValueException.java
#	src/main/java/snomparser/Parser.java
#	src/main/java/snomtasklist/TaskList.java

RunjiaChen committed:
Merge branch 'branch-A-Level-9'

* branch-A-Level-9:
  Add find command

# Conflicts:
#	src/main/java/InputCommands/ListCommand.java
#	src/main/java/SnomParser/Parser.java
#	src/main/java/SnomTaskList/TaskList.java
#	src/main/java/inputcommands/FindCommand.java

RunjiaChen committed:
Add TaskList

RunjiaChen committed:
Merge branch 'branch-A-Level-9'

* branch-A-Level-9:
  Add find command

# Conflicts:
#	src/main/java/inputcommands/FindCommand.java
#	src/main/java/inputcommands/ListCommand.java

RunjiaChen committed:
Merge branch 'master' of https://github.com/RunjiaChen/ip

# Conflicts:
#	src/main/java/inputcommands/ListCommand.java

RunjiaChen committed:
no message

RunjiaChen committed:
Amend merge conflicts

fy17ohhh commented on src/main/java/Snom.java:
```diff
@@ -0,0 +1,43 @@
+import inputcommands.Command;
+import snomexceptions.InvalidCommandException;
+import snomparser.Parser;
+import SnomStorage.TaskStorage;
```
Should package names be lower cases instead?

fy17ohhh commented on src/main/java/Snom.java:
```diff
@@ -0,0 +1,43 @@
+    public Snom() {
+        this.ui = new Ui();
+        this.data = new TaskStorage();
+        this.lst = TaskList.makeTaskList();
```
perhaps use a more comprehensive name instead of `lst`?

fy17ohhh commented on src/main/java/inputcommands/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,83 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+
```
Would it be better to have only one empty line here?

fy17ohhh commented on src/main/java/inputcommands/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,83 @@
+    }
+
+    /**
+     * {@inheritDoc}
+     * In this subclass, we implement date checks and task description checks
```
Might be better to have indentation after `{@inheritDoc}` tag
```suggestion
     * {@inheritDoc}
     *     In this subclass, we implement date checks and task description checks
```

fy17ohhh commented on src/main/java/snomtask/Task.java:
```diff
@@ -0,0 +1,64 @@
+
+    @Override
+    public String toString() {
+        if (done) {
+            return "[X]" + this.name;
+        } else {
+            return "[ ]" + this.name;
+        }
```
Perhaps use ternary operator here to improve tidiness? 🤔

fy17ohhh commented on src/main/java/snomtask/Task.java:
```diff
@@ -0,0 +1,64 @@
+
+    private String name;
+
+    private boolean done;
```
Should boolean names have a prefix such as **is, has, was**?

fy17ohhh commented on src/main/java/snomtasklist/TaskList.java:
```diff
@@ -0,0 +1,100 @@
+import java.util.ArrayList;
+
+public class TaskList {
+    private ArrayList<Task> task_lst;
```
Should variable names be in camelCase instead of having an underline?

fy17ohhh commented on src/main/java/snomui/Ui.java:
```diff
@@ -0,0 +1,44 @@
+import java.util.Scanner;
+
+public class Ui {
+    private Scanner input;
```
perhaps use other names rather than`input` as Scanner Class is input device, not input 🤔

fy17ohhh commented on src/test/java/inputcommands/AddDeadlineCommandTest.java:
```diff
@@ -0,0 +1,51 @@
+public class AddDeadlineCommandTest {
+
+    @Test
+    public void AddDeadlineCommandTest1() {
```
Would it be better to use underscores in test method names? You could follow this format `featureUnderTest_testScenario_expectedBehavior()`

fy17ohhh reviewed (commented):
Please see comments for suggested improvements on coding standards and naming. Overall code quality LGTM, just a few nits to fix.

RunjiaChen committed:
Add GUI to Snom

RunjiaChen committed:
Add Checkstyle

RunjiaChen committed:
Merge branch 'branch-A-CheckStyle'

* branch-A-CheckStyle:
  Add Checkstyle

RunjiaChen committed:
Merge branch 'Branch-Level-10'

* Branch-Level-10:
  Add GUI to Snom

# Conflicts:
#	src/main/java/Snom.java
#	src/main/java/snomparser/Parser.java
#	src/main/java/snomtasklist/TaskList.java

RunjiaChen committed:
Update gradle file for entry point of application

RunjiaChen committed:
Add assertions

RunjiaChen committed:
Merge pull request #1 from RunjiaChen/branch-A-Assertions

Add assertions to the code

RunjiaChen committed:
Change names of methods for Parser Class

There are identical names for methods for the Parser class and the
TaskList class.

Changing the names for the Parser class and TaskList class can improve
code quality, as the method names now provide a clearer idea of what the
method does to the respective classes.

Let's changes the names of addTask methods of Parser to
addTaskToTaskList and addTask respectively.

The reader will be able to
understand that Parser is modifying the instance of TaskList, while
the TaskList is getting modified through the usage of the AddTask
method.

RunjiaChen committed:
Merge pull request #6 from RunjiaChen/branch-A-CodeQuality

Change names of methods for Parser Class

RunjiaChen committed:
Merge branch 'master' of https://github.com/RunjiaChen/ip

* 'master' of https://github.com/RunjiaChen/ip:
  Change names of methods for Parser Class

RunjiaChen committed:
Add detection for duplicate tasks

There is no detection for duplicate tasks entered by the user.

This may cause confusion to the users as they use the list function
and see duplicate tasks.

Let's add checks to detect duplicate tasks within the TaskList.

The checks will check on the behalf of the users for duplicate tasks.

RunjiaChen committed:
Add duplicate task detection

RunjiaChen committed:
Merge branch 'branch-C-DetectDuplicates'

* branch-C-DetectDuplicates:
  Add duplicate task detection
  Add detection for duplicate tasks

RunjiaChen committed:
Add new Exception classes

There is no exception to detect if the user is trying to mark a task
that is completed and unmark a task that is not done.

Let's handle these exceptions by adding a new exception and checking the
status of the task before letting the user mark it as done or undone.

Also add bug fixes for the GUI (The name "Snom" is now correctly
displayed.

Todo: Remove the dead code from the remaining files.

RunjiaChen committed:
Add more exceptions

Users are still able to mark an already completed task as complete.
This may confuse users as some tasks are already done.

Let's add checks to make sure that the user cannot mistakenly mark tasks
that are already done.

RunjiaChen committed:
Merge branch 'branch-A-MoreErrorHandling'

* branch-A-MoreErrorHandling:
  Add more exceptions
  Add new Exception classes

RunjiaChen committed:
Add user guide

RunjiaChen committed:
Merge branch 'branch-A-UserGuide'

* branch-A-UserGuide:
  Add user guide

RunjiaChen committed:
Change png file name

RunjiaChen committed:
Change Gradle Build file

RunjiaChen committed:
Add more JUnit tests

RunjiaChen committed:
Merge branch 'branch-A-MoreTesting'

* branch-A-MoreTesting:
  Add more JUnit tests

RunjiaChen committed:
Add greeting and exit message

RunjiaChen committed:
Merge branch 'FinalTestBranch'

* FinalTestBranch:
  Add greeting and exit message
